FridgeExercise/Refrigerator.cs

```diff
@@ -0,0 +1,249 @@
+			RefrigeratorID = CounterID;
+			CounterID++;
+			Model = model;
+			Color = color;
+			NumberOfShelves = numberOfShelves;
+			Shelves = new List<Shelf>();
+
+		}
+
+		public string ToString(Refrigerator refrigerator)
+		{
+			return "Refrigerator: Model:  " + refrigerator.Model + "  Color: " + refrigerator.Color + "  NumberOfShelves:  "+ refrigerator.NumberOfShelves + "";
+		}
+
+		public double SpaceIsLeftInTheFridge(List<Shelf> shelves)
+		{
+			double currentSpace = 0;
+			foreach (Shelf shelf in shelves)
+			{
+				currentSpace += shelf.CurrentSpaceInShelf;
+			}
+			return currentSpace;
+		}
+		public void PutItemInTheFridge(Item item, List<Shelf> shelves)
+		{
+			foreach (Shelf shelf in shelves)
+			{
+				if (shelf.CurrentSpaceInShelf - item.SpaceItTakes >= 0)
+				{
+
+					shelf.CurrentSpaceInShelf -= item.SpaceItTakes;
+					shelf.Items.Add(item);
+					item.ItemBelongsToShelf = shelf;
+					return;
+				}
+			}
+			Console.WriteLine("There is no  space for this item");
+		}
+		public string TakeOutItem(int ItemId, List<Shelf> shelves)
+		{
+			Item tempItem=null;
+			foreach(Shelf shelf in shelves)
+			{
+				foreach(Item item in shelf.Items.ToList())
+				{
+					if(item.ItemID == ItemId)
+					{
+						tempItem = item;
+						shelf.CurrentSpaceInShelf+=item.SpaceItTakes;
+						shelf.Items.Remove(item);
+						break;
+					}
+				}
+			}
+			if (tempItem == null)
+
+				return "tempItem in TakeOutItem function is null ";
+
+			else
+				return tempItem.ToString(tempItem);
+		}
+		public void FridgeCleaning(List<Shelf> shelves)
+		{
+			foreach (Shelf shelf in shelves)
+			{
+				foreach(Item item in shelf.Items.ToList())
+				{
+					Console.WriteLine(item.ToString(item));
+					if (DateTime.Compare(DateTime.Now, item.ExpiryDate) > 0)
+					{
+						shelf.CurrentSpaceInShelf += item.SpaceItTakes;
+						shelf.Items.Remove(item);
+					}
+				}
+			}
+			
+
+		}
+		public void WhatDoIWantToEat(string type,string kosherType, List<Shelf> shelves)
+		{
+			foreach (Shelf shelf in shelves)
+			{
+				foreach (Item item in shelf.Items)
+				{
+					if (item.Type.ToString()==type && item.KosherType.ToString()==kosherType&& DateTime.Compare(DateTime.Now, item.ExpiryDate)<= 0)
+					{
+						Console.WriteLine(item.ToString(item));
+					}
+				}
+			}
+		}
+		public void SortItemsByDate(List<Shelf> shelves)
+		{
+			List<Item> AllItems= new List<Item>();
+			foreach (Shelf shelf in shelves)
+			{
+				AllItems.AddRange(shelf.Items);
+				
+			}
+
+			AllItems.Sort((x, y) => DateTime.Compare(x.ExpiryDate, y.ExpiryDate));
+			foreach (Item item in AllItems)
+			{
+				Console.WriteLine(item.ToString(item));
+			}
+		}
+		public void SortShelvesBySpace(List<Shelf> shelves)
+		{
+			List<Shelf> sortedShelvesList= shelves.OrderByDescending(x => x.CurrentSpaceInShelf).ToList();
+			foreach (Shelf shelf in sortedShelvesList)
+			{
+				Console.WriteLine(shelf.ToString(shelf));
+			}
+		}
+
+		public void GoingShopping(List<Shelf> shelves)
```

Comment:
```suggestion
		public void PrepareForShopping(List<Shelf> shelves)
```

That name, also, is more a call for action